runtime/parser/schema/rillyaml.schema.json

```diff
@@ -0,0 +1,129 @@
+          "description": "Defines project-wide default settings for models. Unless overridden, individual models will inherit these defaults"
+        },
+        "metrics_views": {
+          "$ref": "./resource.schema.json#/$defs/metrics_view/$defs/metrics_view_properties",
+          "description": "Defines project-wide default settings for metrics_views. Unless overridden, individual metrics_views will inherit these defaults"
+        },
+        "explores": {
+          "$ref": "./resource.schema.json#/$defs/explore/$defs/explore_properties",
+          "description": "Defines project-wide default settings for explores. Unless overridden, individual explores will inherit these defaults"
+        },
+        "migrations": {
+          "$ref": "./resource.schema.json#/$defs/migration/$defs/migration_properties"
+        },
+        "features": {
+          "oneOf": [
+            {
+              "type": "object",
+              "additionalProperties": {
+                "type": "boolean"
+              }
+            },
+            {
+              "type": "array",
+              "items": {
+                "type": "string"
+              }
+            }
+          ],
+          "description": "Enables or disables experimental or optional features using key-value pairs, where the key is the feature name and the value is a boolean"
+        },
+        "public_paths": {
+          "type": "array",
+          "description": "List of file or directory paths to expose over HTTP. Defaults to ['./public']",
+          "items": {
+            "type": "string"
+          }
+        },
+        "ignore_paths": {
+          "type": "array",
+          "description": "A list of file or directory paths to exclude from parsing. Useful for ignoring extraneous or non-Rill files in the project",
+          "items": {
+            "type": "string"
+          }
+        },
+        "mock_users": {
+          "type": "array",
+          "description": "A list of mock users used to test dashboard security policies within the project",
+          "items": {
+            "type": "object",
+            "properties": {
+              "email": {
+                "type": "string",
+                "description": "The email address of the mock user. This field is required"
+              },
+              "name": {
+                "type": "string",
+                "description": "The name of the mock user."
+              },
+              "admin": {
+                "type": "boolean",
+                "description": "Indicates whether the mock user has administrative privileges"
+              },
+              "groups": {
+                "type": "array",
+                "description": "An array of group names that the mock user is a member of",
+                "items": {
+                  "type": "string"
+                }
+              }
+            },
+            "required": [
+              "email"
+            ]
```

Comment:
These are the main properties, but we also allow arbitrary values attributes here. Does the schema fail if an arbitrary attribute is specified or does it gracefully handle unknown attributes? 

Is there a way to configure whether or not unknown properties are allowed on a per-type basis?

Reply:
Great! I agree that sounds safest